File: random/1638b.py
import sys
t = int(input())
for i in range(t):
    n = int(input())
    arr = sys.stdin.readline().split()
    er = False
    odd = 0
    even = 0
    for i in arr:
        q = int(i)
        if er:
            break
        if q % 2:
            if odd and q < odd:
                er = True
            odd = q
        else:
            if even and q < even:
                er = True
            even = q
    if er:
        print('NO')
    else:
        print('YES')

File: random/1638b.cpp
#include <iostream>
using namespace std;

int main(){
	int t, n, a, odd=0, even=0;
	cin >> t;
	bool er = false;
	while(t--){
		cin >> n;
		odd = 0;
		even = 0;
		er = false;
		while(n--){
			cin >> a;
			if (a % 2 == 0){
				if (even != 0){
					
					if (a < even){
						er = true;
					} 
				}
				even = a;
			}
			else{
				if (odd != 0){
					if (a < odd){
						er = true;
					} 
				}
				odd = a;
			}
		}
		if (er) cout << "NO" << endl;
		else cout << "YES" << endl;
	}

}